src/configuration/get_kata_list/in_memory_kata_list.c

```diff
@@ -0,0 +1,33 @@
+#include "get_kata_list.h"
+#include "katas/katas.h"
+#include "string.h"
+
+const static kata_t KATA_LIST[] = {
+        {.name={.str="intro"}, .path={.str="katas/intro.c"}},
+        {.name={.str="success"}, .path={.str="katas/success.c"}},
+        {.name={.str="compilation error"}, .path={.str="katas/compilation_error.c"}},
+        {.name={.str="run error"}, .path={.str="katas/run_error.c"}},
+        {.name={.str="test error"}, .path={.str="katas/test_error.c"}},
+};
```

Comment:
So that replaces the yml, and it is the part that we're going to update whenever we add katas right ?

If so, with:
- all katas being in katas/*,
- the name being the file name without the .c and with " " instead of "_"

Couldn't we get rid of this, if there is a way to retrieve all files in katas/* in a predictable order ?

If this is possible, it would avoid potential runtime errors (the files not being in sync with this hardcoded list) and make the process of creating and changing katas smoother.

What do you think ?

Reply:
I think you're right. I change it 👍

Reply:
Great, thanks !